### ISSUE TITLE
Building static protobuf on Windows forces linking to static MSVC runtime.

### ISSUE THREAD

**Issue by ambrop72** [protocolbuffers/protobuf#9186]
When protobuf is built as a static library on Windows using CMake, it is linked to the static MSVC runtime. This is not compatible with an application that links to the dynamic MSVC runtime, since all objects linked into the application must use the same runtime. This issue is present when using CMake >=3.15 due to the [logic in CMakeLists](https://github.com/protocolbuffers/protobuf/blob/7ccf4d8f67878a6ceb2184df279478cb3314372b/cmake/CMakeLists.txt#L184). The `protobuf_MSVC_STATIC_RUNTIME` option is not respected when using CMake >=3.15. I am forced to patch CMakeLists for my application. It would be great if the logic was fixed to respect this option.

One way to do this is to change
```
set(CMAKE_MSVC_RUNTIME_LIBRARY MultiThreaded$<$<CONFIG:Debug>:Debug>)
```
to
```
if (protobuf_MSVC_STATIC_RUNTIME)
  set(CMAKE_MSVC_RUNTIME_LIBRARY MultiThreaded$<$<CONFIG:Debug>:Debug>)
else ()
  set(CMAKE_MSVC_RUNTIME_LIBRARY MultiThreaded$<$<CONFIG:Debug>:Debug>DLL)
endif ()
```

**Comment by mloskot** [comment]
I confirm it, I've just hit the same bug.

**Comment by polizi** [comment]
This problem can be solved by `    set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
    set(protobuf_MSVC_STATIC_RUNTIME   OFF)`

**Comment by acozzette** [comment]
This should be fixed with #9437.
